### PR TITLE
Applicant and Application Endpoints and Test Cases

### PR DESCRIPTION
**Pull request by simpslandyy**
please review :) 

### NEW
/applicants_controller ||  added index and show methods
/applications_controller || added index and show methods
/applicants_controller_spec || added test cases for applicants controller
/applications_controller_spec || added test cases for application controller

### REVIEW

Please check that the test cases run.
Check that /applicants and /applications return a list of applicants and a list of applications
Check that /applicants/:id and /applications/:id return a single applicant and application
Check non integer ids throw a 404

### CODE REVIEW

**Comment by michellemtchai** [comment]
````
ApplicationsController
  GET #index
    when passed an applicant ID
      list applications given an integer applicant ID
      list applciations given a non-integer applicant ID
    when not passed an applicant ID
      list applications
  GET #show
    when passed an integer ID
      returns the application associated to integer ID of an applicant
    when passed a non-integer ID
      returns 404

Finished in 0.16046 seconds (files took 1.48 seconds to load)
5 examples, 0 failures
````

````
ApplicantsController
  GET #index
    list applicants
  GET #show
    when passed a non-integer ID
      returns 404
    when passed an integer ID
      returns 200
      returns 200

Finished in 0.05375 seconds (files took 1.52 seconds to load)
4 examples, 0 failures
````
Both sets of test passed. Are these the expected results?

**Comment by michellemtchai** [comment]
The results for the `/applicants` returned an array with attributes:
- id
- utorid
- student_number
- first_name
- last_name
- email
- phone 
- address
- created_at
- updated_at

**Comment by michellemtchai** [comment]
The results for the `/applications` returned an array with attributes:
- id
- applicant_id
- app_id
- round_id
- ta_experience
- research
- comments
- availability
- degrees
- work_experience
- hours_owed
- pref_session
- pref_campus
- deferral_status
- deferral_reason
- appointment_number
- created_at
- updated_at

The `preferences` attribute is not visible in this view. Is the the expected result?

**Comment by michellemtchai** [comment]
The results for the `/application/:id` returned an object with the same attributes as that of `/application/:id` with the addition of one more attribute, `preferences`. This new attribute contains an array of objects with attributes:
- id
- application_id
- position_id
- rank
- created_at
- updated_at

The use of non-integer/non-existent `id` does throw a `status: 404`.

**Comment by michellemtchai** [comment]
The results for the `/applicants/:id` returned an object with the same attributes at the ones from `/applicants`. There are no extra attributes returned.

The use of non-integer/non-existent `id` does throw a `status: 404`.

File: spec/controllers/applicants_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+      let(:parsed_body) { JSON.parse(response.body) }
+
+      it "returns 200" do
+        get :show, params: { id: applicant.id }
+        expect(response.status).to eq(200)
+        expect(response.body).not_to be_empty
+      end
+
+      it "returns 200" do
```
Comment (freeatnet) [inline spec/controllers/applicants_controller_spec.rb]: Probably not the right description :D

**Review by freeatnet — APPROVED** [review]

File: spec/controllers/applicants_controller_spec.rb (hunk 1 above)
Comment (simpslandyy) [reply spec/controllers/applicants_controller_spec.rb]: Oh you're right! Thank you for the catch.

**Comment by simpslandyy** [comment]
@freeatnet - I've fixed the description for the applicants test
@michellemtchai - yes those are the correct results for the endpoints and the tests.

**Review by michellemtchai — APPROVED** [review]

**Pull request merged**